gnomad_qc/v4/create_release/create_de_novo_release.py

```diff
@@ -42,304 +31,210 @@
-            n_de_novo=ht[f"n_de_novos_{g}"],
-            parents=hl.struct(
-                AC=ht[f"ac_parents_{g}"],
-                AN=ht[f"an_parents_{g}"],
-            ),
-            children=hl.struct(
-                AC=ht[f"ac_children_{g}"],
-                AN=ht[f"an_children_{g}"],
-            ),
+    mt = mt.annotate_rows(prior=priors_ht[mt.row_key].freq[0].AF)
+    mt = mt.select_entries("GT", "AD", "DP", "GQ", "PL", "adj")
+
+    tm = hl.trio_matrix(mt, ped, complete_trios=True)
+    tm = tm.transmute_cols(is_xx=tm.is_female)
+    tm = tm.checkpoint(new_temp_file("trio_matrix", "mt"))
+
+    ht = tm.entries()
+
+    ht = ht.annotate(
+        de_novo_call_info=default_get_de_novo_expr(
+            locus_expr=ht.locus,
+            alleles_expr=ht.alleles,
+            proband_expr=ht.proband_entry,
+            father_expr=ht.father_entry,
+            mother_expr=ht.mother_entry,
+            is_xx_expr=ht.is_xx,
+            freq_prior_expr=ht.prior,
         )
-        for g in ["adj", "raw"]
-    }
-
-
-def custom_joint_freq_select(ht: hl.Table, **_) -> Dict[str, hl.expr.Expression]:
-    """
-    Select joint freq fields for release.
-
-    :param ht: Joint freq Hail Table.
-    :return: Select expression dict.
-    """
-    selects = {
-        f"{data_type}_freq": ht[data_type].freq
-        for data_type in ["exomes", "genomes", "joint"]
-    }
-    selects["exomes_qual_hists"] = ht.exomes.histograms.qual_hists
+    )
 
-    return selects
+    ht = ht.filter(ht.de_novo_call_info.is_de_novo).naive_coalesce(1000)
+    return ht
 
 
-def custom_joint_freq_select_globals(ht: hl.Table) -> Dict[str, hl.expr.Expression]:
+def aggregate_and_annotate_de_novos(ht: hl.Table) -> Tuple[hl.Table, hl.Table]:
     """
-    Select joint freq globals for release.
-
-    :param ht: Joint freq Hail Table.
-    :return: Select expression dict
-    """
-    return {
-        f"{data_type}_freq_globals": ht[f"{data_type}_globals"].select(
-            "freq_meta", "freq_index_dict", "freq_meta_sample_count"
-        )
-        for data_type in ["exomes", "genomes", "joint"]
-    }
+    Aggregate and annotate de novo calls.
 
+    This step get two sets of de novo calls:
+       - de novos at all confidence levels
+       - filtered high-confidence coding de novos
 
-def custom_info_select(ht: hl.Table, **_) -> Dict[str, hl.expr.Expression]:
+    :param ht: De novo calls Table.
+    :return: Aggregated and annotated Tables.
     """
-    Select fields for info Hail Table annotation in de novo release.
+    # Get needed annotation Tables.
+    filters_ht = final_filter(all_variants=True, only_filters=True).ht()
+    freq_ht = public_release("exomes").ht()
+    vep_ht = get_vep(data_type="exomes").ht()
 
-    :param ht: Info Hail Table.
-    :return: Select expression dict.
-    """
-    return {"allele_info": ht.allele_info.drop("nonsplit_alleles")}
+    ht = ht.filter(~hl.is_missing(ht.de_novo_call_info.confidence))
```

Comment:
this filter is in place to remove variants that were candidate de novos (per `ht.de_novo_call_info.is_de_novo`) but failed for another reason, right? do you think users would find it useful to see these variants in the Hail Table with all de novos?

Reply:
yeah, if we keep them, we'd have to calculate the AC across these sites (sites that were candidate de novos but failed for some other quality reason) and subtract them to report the various AC (AC_all, high confidence AC, etc).

could you ask in your ATGU presentation whether people think it would be useful to retain these variants? I suspect the ATGU community will include some users of this new feature, so they would be the best people to ask about this

Reply:
sorry missed your comment two weeks ago. given Mark and Konrad's thoughts (we should only release high-medium confidence de novos), this thread is no longer relevant, so feel free to resolve